nodemapper/src/gui/Builder/components/HighlightedJSON.tsx

```diff
@@ -90,61 +326,144 @@ const HighlightedJSON = (props: HighlightedJSONProps) => {
+            keylist[keylist.length - 1],
+          );
+          delete parent[":" + key];
+        }
+        dispatch(builderUpdateNode(newnode));
+      };
+
       return (
-        <div key={key} className="line">
-          <span className="key">{key}:</span>
+        <div style={{ cursor: "default" }} key={key} className="line">
           {isProtectedValue ? (
-            <span
-              className="protected"
-              style={{
-                display: "inline-block",
-                height: "25px",
-                minWidth: "150px",
-              }}
-            >
-              {value}
+            <span>
+              <span className="key">{label}:</span>
+              <span
+                className="protected"
+                style={{
+                  display: "inline-block",
+                  height: "25px",
+                }}
+              >
+                {value}
+              </span>
             </span>
-          ) : isSimpleValue ? (
-            <span
-              className={valueType}
-              style={{
-                display: "inline-block",
-                height: "25px",
-                minWidth: "150px",
-              }}
-            >
-              {valueType === "boolean" ? (
-                <EasyEdit
-                  type={Types.SELECT}
-                  onHoverCssClass="easyedit-hover"
-                  saveButtonLabel={<FontAwesomeIcon icon={faCheck} />}
-                  cancelButtonLabel={<FontAwesomeIcon icon={faTimes} />}
-                  value={value}
-                  options={[
-                    { label: "true", value: true },
-                    { label: "false", value: false },
-                  ]}
-                  onSave={(value) => {
-                    setValue(value);
-                  }}
-                  saveOnBlur={true}
-                />
-              ) : (
+          ) : isParameterConnected ? (
+            <span>
+              <span className="key">{label}:</span>
+              <span
+                className="linked"
+                style={{
+                  display: "inline-block",
+                  height: "25px",
+                }}
+              >
                 <EasyEdit
                   type={Types.TEXT}
+                  style={{ cursor: "text" }}
                   onHoverCssClass="easyedit-hover"
-                  saveButtonLabel={<FontAwesomeIcon icon={faCheck} />}
-                  cancelButtonLabel={<FontAwesomeIcon icon={faTimes} />}
-                  value={value}
+                  value={parameterValue}
                   onSave={(value) => {
-                    setValue(value);
+                    return;
                   }}
-                  saveOnBlur={true}
+                  allowEdit={false}
                 />
-              )}
+              </span>
+              <ConnectParameter connectParameter={connectParameter} />
+            </span>
+          ) : isSimpleValue ? (
```

Comment:
```suggestion
          ) : isSimpleValue && (
```
and remove `: null ` line 461